### PR TITLE
Kilo Chapel Revamp (Pubby Style)

### PR DESCRIPTION
**Pull request by shelteredfreak**
## About The Pull Request

First PR pog.

This map edit was made because I was dissatisfied with what the Kilo chapel had and yearned for more space... And perhaps a nostalgic aspect to it? So with the help of MrMelbert and San7890, this was finished. 

Monas Final

A complete revamp of the chapel, moving it off of the main station and shoving it into an asteroid. It is finished pretty much. Though the Observatory area of the Monastery could use more decorations, and definitely could add more stuff onto the tables. Everything essential is here.

But you might be wondering what happened to where the chapel was...

Chapel Final

Big thanks to MrMelbert for doing up this place for me, all of the stuff for the garden and hallway was his work. 

It is replaced with a public garden, something Kilo was missing. With a transit tube where the mass driver was, leading to the Monastery.

Also shoutout for San7890 for helping me figure a lot of this stuff out.

Oh, and since the cremator is off station, there is a new one where the Sec hardsuits were, moving them into the armory.

If anything needs fixed, feel free to ping me on discord to let me know. This is my first time contributing to any code ever so it helps to know what I did wrong.

## Why It's Good For The Game

Kilo's chapel is... to put it bluntly, a cabin with an altar. It is tiny, cramped, and no room for religious activities! So I thought it would be better if the whole thing was moved off station, like Pubby's Chapel was, for more room!

## Changelog

:cl: Shelteredfreak, MrMelbert, and San7890
del: Removed Kilo's chapel in favor of a monastery off of the main station.
add: Added a new monastery on Kilo's Northwestern asteroid.
add: Added a security cremator to sec in place of the hardsuits, hardsuits moved to armory.
add: Added a public garden in the Chapel's place, complete with cow and chicken (Done by MrMelbert).
/:cl:

### CODE REVIEW

**Comment by JohnFulpWillard** [comment]
why not just bring pubby back instead

**Comment by shelteredfreak** [comment]
> why not just bring pubby back instead

As much as I would love that option, I don't think they'd let me....

**Comment by Jolly-66** [comment]
> why not just bring pubby back instead

Fulp haven't you done work on it? Couldn't we try to get Pubby back in rotation?

**Comment by JohnFulpWillard** [comment]
> > why not just bring pubby back instead
> 
> Fulp haven't you done work on it? Couldn't we try to get Pubby back in rotation?

yeah its, for the most part, up-to-date on the map depot. very little if any work is needed.

**Comment by MrMelbert** [comment]
> why not just bring pubby back instead

Maptainers have already said multiple times they aren't interested in swapping kilo and pubby back and forth

**Comment by Serxule** [comment]
pubby chapel was one of the main things i hated about it, this is very not epic.

**Comment by shelteredfreak** [comment]
> pubby chapel was one of the main things i hated about it, this is very not epic.

A wise man once said "Cope." Because a monastery is badass, and bigger is better.

**Comment by Serxule** [comment]
> > pubby chapel was one of the main things i hated about it, this is very not epic.
> 
> A wise man once said "Cope." Because a monastery is badass, and bigger is better.

pubby monastery is the most cramped and maze-like building i know of

**Comment by shelteredfreak** [comment]
> > > pubby chapel was one of the main things i hated about it, this is very not epic.
> > 
> > 
> > A wise man once said "Cope." Because a monastery is badass, and bigger is better.
> 
> pubby monastery is the most cramped and maze-like building i know of

Funny thing is I have never actually played on Pubby as a chaplain before in my life. I don't even know what it's like. But I think it looks poggers, plus Kilo's current chapel looks like garbaggio. Far more cramped than what I've seen of Pubby chapel.

**Comment by SuperNovaa41** [comment]
holy crap this is awesome

**Comment by shelteredfreak** [comment]
> holy crap this is awesome

thanks SuperNovaa41 (maintainer of TG, and Voidcrew???)

Soon (when I figure out how to make a new map with Dream Maker) I will go to voidcrew and make funny clockwork ship.

**Comment by Serxule** [comment]
> > > > pubby chapel was one of the main things i hated about it, this is very not epic.
> > > 
> > > 
> > > A wise man once said "Cope." Because a monastery is badass, and bigger is better.
> > 
> > 
> > pubby monastery is the most cramped and maze-like building i know of
> 
> Funny thing is I have never actually played on Pubby as a chaplain before in my life. I don't even know what it's like. But I think it looks poggers, plus Kilo's current chapel looks like garbaggio. Far more cramped than what I've seen of Pubby chapel.

that is true current kilo chapel is ass but i dont like pubby related things

**Comment by Farquaar** [comment]
While I really like Pubby's old monastery, I'm not sure if I want to see it on Kilo. To me, Kilo's appeal is in its small size. Kilo keeps people close together, thereby maximizing the number of interactions with your fellow crew. The monastery kinda separates people out from the station, which kinda goes against what Kilo is supposed to be.

TL;DR I'm not entirely against the idea of a monastery on Kilo, but this one might be a tad too big.

**Comment by Fryzzie** [comment]
whilst i like the new chapel, There really probably shouldn't be a robotics exofab and circuit imprinter off the main station.

**Comment by shelteredfreak** [comment]
>To me, Kilo's appeal is in its small size. Kilo keeps people close together, thereby maximizing the number of interactions with your fellow crew. The monastery kinda separates people out from the station, which kinda goes against what Kilo is supposed to be.
> 
> Basically I'm not entirely against the idea of a monastery on Kilo, but this one might be a tad too big.

 I understand what Kilo is supposed to be and I don't really approve of it. Forcing interactions between players who probably don't want to is not healthy. The reason I expanded it off to a monastery is to encourage player creativity, which is something that Kilo does not allow.
 
 Kilo's cramped hallways and tiny building areas is perfect for forcing interactions, yes. But there is no room for players to experiment with machines, designs or whatever they want. The goal of expanding the monastery is to give attention to an oft overlooked job: The Chaplain. Whenever there isn't cult they can redecorate and refit the monastery to their god, or better yet experiment with some designs.
 
 TL:DR: Kilo forces interactions, bad. I want to encourage player creativity, good.

**Comment by shelteredfreak** [comment]
> whilst i like the new chapel, There really probably shouldn't be a robotics exofab and circuit imprinter off the main station.

The exofab and imprinter are unlinked to the ore silo, meaning it takes some effort to get them working. Although I did notice from this comment that the table in that room is an abductor table and can be used for Alien Alloy, so that is something.

**Comment by shelteredfreak** [comment]
> that is true current kilo chapel is ass but i dont like pubby related things

> While I really like Pubby's old monastery, I'm not sure if I want to see it on Kilo.

Although I put Pubby in the title, because the idea of a monastery is taken from it. Don't let any feelings you have against Pubby carry onto this. It was only a style option, I made this entire thing myself (except the hallway leading to it). Unless you would rather I'd just copy paste the entire Pubby Monastery and be done with it?

**Comment by Fikou** [comment]
it shouldnt have a circuit imprinter, probably not the ouija board and security shouldnt have a cremator

**Comment by shelteredfreak** [comment]
> it shouldnt have a circuit imprinter, probably not the ouija board and security shouldnt have a cremator

The circuit imprinter is the offstation one, meaning you need to sync it up. 

The spirit board is never used, and chaplains are intended to commune with spirits, due to their camera obscura. 

And I was advised to make a cremator in security since the current cremator was moved off into the monastery

**Comment by MrMelbert** [comment]
image

hmm

**Comment by shelteredfreak** [comment]
> image
> 
> hmm

guh, let me fix that

**Comment by Fikou** [comment]
> The circuit imprinter is the offstation one, meaning you need to sync it up. The exofab is the maint one, also needing synced.

what do you mean sync it up i just need to grab some glass and iron and i have a ripley lol

> The spirit board is never used, and chaplains are intended to commune with spirits, due to their camera obscura.

how does spirit photo equal intended to commune? and the spirit board is never used, probably for a reason. that being we want to limit ghost/crew interaction

> And I was advised to make a cremator in security since the current cremator was moved off into the monastery

security having access to the round removal tool for free is kinda lame, couldn't you hook the cremator up to for example the morgue? its already next to where chapel used to be and it serves mostly the same purpose as the morgue rooms in the chapel

**Comment by shelteredfreak** [comment]
> what do you mean sync it up i just need to grab some glass and iron and i have a ripley lol

The main purpose of these rooms is to act as supplementary to certain religions, chaplains are more than likely to make stuff of their own here if they need it. But newer chaplains can explore these rooms and take inspiration. Without the fabricator or imprinter I might as well remove the entire room and make it more bland. They are meant to mimic Kilo's maint where you can find various items and stuff. Like the russian soldiers room where you can literally just get guns. 

Perhaps I should put a hostile mob in there to prevent use instead?

> how does spirit photo equal intended to commune? and the spirit board is never used, probably for a reason. that being we want to limit ghost/crew interaction

The spirit board requires several people to commune anyways. There is no way to get it easily. This can be used for great roleplay moments of chaplains gathering people to the monastery and then communing with the spirits. It also works the exact same to an IRL ouija board, they can only spell out one letter at a time.

It is vastly underpowered and I don't see why you are complaining.

> security having access to the round removal tool for free is kinda lame, couldn't you hook the cremator up to for example the morgue? its already next to where chapel used to be and it serves mostly the same purpose as the morgue rooms in the chapel

No. Because then all of the medical doctors have access to it. Then a med doctor could easily chloral you and cremate you without you having a chance. It is meant to be kept in a secure area for ling removals and the like. 

Besides, if security wants you perma dead, they can do that easily with access and weapons. Putting a cremator in security isnt gonna make it any more easier to perma round remove you.

**Comment by Son-of-Space** [comment]
I like this PR a lot! This little piece of Kilo is a great way to incrementally build your mapping skills. A few pieces of feedback:

1. One of the things that made the PubbyStation chapel tolerable was its ferry that fed to and from the station. You could effectively transport people you had in custody (or had kidnapped), move bulk crates, etc. I wouldn't suggest adding one in this PR to avoid scope creep, but I HIGHLY suggest looking into the quagmire that is shuttle code and working to implement a ferry system here as well sometime in the near future. It will heavily improve the accessibility and thus utilization of the space you've made.

2. I loved PubbyStation's Chapel (So much that I turned it into a purchasable shuttle just for the sheer size of it.) That comically massive size was one of its downfalls, though. I would strongly suggest looking into how you can efficiently utilize the space you've built and size down the area of the chapel to help keep it from clashing with Kilo's current aesthetic. My challenge to you to help you become a strong mapper would be to remove 30-40 tiles of space that you aren't using effectively.

My big tip here: Mapping is not painting. What looks visually pleasing in the map editor may not be pleasing to play or use. Think about what is there to make it look fluffy, and what is there to make it good for gameplay. Combine the two and kill your darlings. Also think about the standard view-range for a player.

3. The mapdiff renders might be showing it wrong, but it looks like you have boards over reinforced walls. It's fine to have these be doors instead of walls, and having the boards under doors is what they are intended for.

4. I would avoid adding stair tiles to the design. While they're intended to show height difference, we want their use to stay consistent throughout the game and the way they are being used here could be seen as tacky once the illusion fades.

5. Having the double reinforced windows for the tube on the south side makes sense because there's a maintenance walkway there, but having them on the top creates an easy way to trap someone. Try using two sets of directional windows on the top entryway instead.

6. I have to agree with Fikou. Security doesn't need a cremator. I would suggest putting it on the opposite side of the entryway to the chapel instead of next to medbay if you really want to have a cremator station-side, but if there's one on the chapel area, you don't really need two.

7. The exofab IS probably a bit much, though I understand the techpriest aesthetic you're going for. I would ask what you think you could do to promote that same aesthetic and encourage that style without giving them access to an entire arsenal after a few trips to the ORM.

I have no comment on the spirit board. I think that giving the crew access to metaknowledge like ghosts should not be available roundstart, but I haven't used it so I can't give clear guidance.

**Comment by shelteredfreak** [comment]
> I like this PR a lot! This little piece of Kilo is a great way to incrementally build your mapping skills. A few pieces of feedback:
> 
> 1. One of the things that made the PubbyStation chapel tolerable was its ferry that fed to and from the station. You could effectively transport people you had in custody (or had kidnapped), move bulk crates, etc. I wouldn't suggest adding one in this PR to avoid scope creep, but I HIGHLY suggest looking into the quagmire that is shuttle code and working to implement a ferry system here as well sometime in the near future. It will heavily improve the accessibility and thus utilization of the space you've made.

I can try looking into it, because I did actually consider adding a ferry but thought that it would be too much for right now.

> 2. I loved PubbyStation's Chapel (So much that I turned it into a purchasable shuttle just for the sheer size of it.) That comically massive size was one of its downfalls, though. I would strongly suggest looking into how you can efficiently utilize the space you've built and size down the area of the chapel to help keep it from clashing with Kilo's current aesthetic. My challenge to you to help you become a strong mapper would be to remove 30-40 tiles of space that you aren't using effectively.

At current, even the main area is excessively small compared to what I would like, but I kept it to its current size so it could still have SOME room to build and design it how you'd like.

> My big tip here: Mapping is not painting. What looks visually pleasing in the map editor may not be pleasing to play or use. Think about what is there to make it look fluffy, and what is there to make it good for gameplay. Combine the two and kill your darlings. Also think about the standard view-range for a player.

Generally this is just going to be a one-off PR for me, because I don't plan on making any other mapping changes (on TG), simply because I don't have the motivation or reason to do it. I started this because I was a chaplain main who wanted a better chapel.

> 3. The mapdiff renders might be showing it wrong, but it looks like you have boards over reinforced walls. It's fine to have these be doors instead of walls, and having the boards under doors is what they are intended for.

That is a falsewall covered by the boards, I could _maybe_ switch it out for a door but the idea was to make it look like it was intended to not be here.

> 4. I would avoid adding stair tiles to the design. While they're intended to show height difference, we want their use to stay consistent throughout the game and the way they are being used here could be seen as tacky once the illusion fades.

I wanted to experiment with the stairs because I rarely see them used anywhere else, plus if you look at where the stairs are planned and how the rooms are setup, you can figure out that the actual place would be shaped somewhat like a castle. I don't feel like it would be a necessity to removes these, and if players want more space they can chop down all the stuff in the way and expand. That was the plan anyways.

> 5. Having the double reinforced windows for the tube on the south side makes sense because there's a maintenance walkway there, but having them on the top creates an easy way to trap someone. Try using two sets of directional windows on the top entryway instead.

I'm not sure I entirely know what you mean? But I think I do, I could maybe switch those out for directional reinforced windows but it drastically reduces the integrity of that area. It was kinda already like this before but I simply made it into a transit tube area.

> 6. I have to agree with Fikou. Security doesn't need a cremator. I would suggest putting it on the opposite side of the entryway to the chapel instead of next to medbay if you really want to have a cremator station-side, but if there's one on the chapel area, you don't really need two.

I would rather remove the cremator from sec and let there only be one on the chapel if I had a choice, but the truth remains that changelings will need to be dealt with one way or another. But in any case, putting it anywhere unsecure is a major risk as the cremator is akin to an SM, it can instantly delete anything that goes inside. If you want I could stick it somewhere in command, but that would be pushing it honestly.

> 7. The exofab IS probably a bit much, though I understand the techpriest aesthetic you're going for. I would ask what you think you could do to promote that same aesthetic and encourage that style without giving them access to an entire arsenal after a few trips to the ORM.

I originally made this room because Delta and Meta have exofabs in maint, but upon reviewing the areas, they do not have circuit imprinters. I'll remove the circuit imprinter since it has caused this much dispute.

> I have no comment on the spirit board. I think that giving the crew access to metaknowledge like ghosts should not be available roundstart, but I haven't used it so I can't give clear guidance.

The spirit board requires the room to be pitch black and 6 or so people to gather around it, have fun gathering the crew to talk to the spirits one letter at a time. I think it needs more attention, because magic eightballs and camera obscuras already exist along with it. But as soon as its a ouija board everyone goes insane.

**Review by Maurukas — CHANGES_REQUESTED** [review]
First, your PR appears to have picked up some commits you are not intending. This is reverting changes made to code/modules/tgs/v3210/api.dm, and is messing with the gbp balances file directly. Please drop these changes from the PR - coding-general should be able to assist if you are not familiar with how to do this. You should be able to interactively rebase and drop those commits.

Secondly, this looks excellent. Although the chapel itself is a deviation from Kilo's standards, the rest of the mapping fits the themes and style of Kilo extremely well - and that's not easy to do. I'm willing to give this a go.

Now, onto actual reviewing:

<details>
<summary> There are 64 active turfs along the transit tube and utilities connection </summary>
Please make sure you are using airless turf subtypes in areas exposed to space.

image
image
</details>



<details>
<summary> Layering issue with the pressure tank </summary>
They only connect on the primary layer because they are awful.

image

</details>
<details>
<summary> It's possible to get trapped by firelocks here </summary>
If you move the slim firelocks out one tile further that should handle it.

image
</details>


<details>
<summary> Notice board flipped the wrong way here </summary>

image
</details>

<details>
<summary> Suggestion: The power draw here is very high, break this area up? </summary>
The draw on the chapel APC is approximately 6kw, mostly due to lighting. This is higher than Kilo's atmospherics, and depletes power within 5 minutes of roundstart. I would suggest using /area/service/chapel/asteroid or /area/service/chapel/dock to split the area and divide the power draw between multiple APCs.

image
image

**Comment by shelteredfreak** [comment]
fug u bot
